src/coreclr/jit/importer.cpp

```diff
@@ -4515,13 +4515,37 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
             case NI_System_Math_Log:
             case NI_System_Math_Log2:
             case NI_System_Math_Log10:
-#ifdef TARGET_ARM64
+#if defined TARGET_ARM64 || defined TARGET_XARCH
             // ARM64 has fmax/fmin which are IEEE754:2019 minimum/maximum compatible
             // TODO-XARCH-CQ: Enable this for XARCH when one of the arguments is a constant
             // so we can then emit maxss/minss and avoid NaN/-0.0 handling
             case NI_System_Math_Max:
             case NI_System_Math_Min:
+            {
+#if defined FEATURE_HW_INTRINSICS && defined TARGET_XARCH
+
+                if (compExactlyDependsOn(InstructionSet_SSE) && callType == TYP_FLOAT)
```

Comment:
Is it right that I check for TYP_FLOAT, which makes double go to the NI_Math_Min/Max path?

Yes, it is not 100% safe, so I have to implement something like this:
https://github.com/dotnet/runtime/blob/6256adf64fcc157ff1db8a33ed30bd70a0ddb68b/src/coreclr/jit/valuenum.cpp#L5868-L5882
but with GT_HWINTRINSIC nodes. don't i?

Reply:
> Is it right that I check for TYP_FLOAT, which makes double go to the NI_Math_Min/Max path?

They should go down the same path, but only differing on which `HWIntrinsic` they pick. `TYP_FLOAT` uses `NI_SSE_*` and `TYP_DOUBLE` uses `NI_SSE2_*`

> Yes, it is not 100% safe, so I have to implement something like this:

No, that logic is for if both inputs are constant. For this case you need at least 1 (but not necessarily both) to be constant. You can check if an operand is a floating-point constant using `op->IsCnsFltOrDbl()`. You can check its underlying value using `op->AsDblCon()->gtDconVal`.

`op->IsFloatPositiveZero()` in particular lets you check for exactly `+0.0` and returns false for `-0.0`, there is not currently any `op->IsFloatNaN()` or similar to check for `NaN` nor any `IsFloatNegativeZero()` to check for exactly `-0.0`. Noting that these in particular are special because `NaN != NaN` and `+0.0 == -0.0`, so they need bitwise checks rather than regular equality checks. Normally you'd just compare `gtDconVal == value`.

If neither input is constant you likely don't want to import as an intrinsic and want to instead leave it as a call so inlining can still happen, so you may want to peek the initial stack values and only pop them if you know 1 or both are constant. This will miss some optimizations where an input becomes a constant "later" (such as in morph) but its much simpler than trying to update the JIT to emit an optimized intrinsic sequence for the fallback case.

Reply:
@tannergooding @jakobbotsch 
I made changes in the code. It seems like It covers all cases that we are concerned of, however spmi gives me bad results:
[diff_summary.13.md](https://github.com/dotnet/runtime/files/8733676/diff_summary.13.md)
[superpmi.46.log](https://github.com/dotnet/runtime/files/8733710/superpmi.46.log)
I can't figure out why there are compilation failures. What have I missed?

Reply:
@tannergooding @jakobbotsch Hey everyone. I kindly remind about my troubles, just in case you missed it. Could you, please look at the PR, I'm obviously missing something that stops the tests from passing successfully.